amp/WEB-INF/src/org/digijava/module/aim/util/ActivityVersionUtil.java

```diff
@@ -351,316 +351,323 @@ public static AmpActivityVersion initializeActivity(AmpActivityVersion act) thro
+                                    // previous iteration.
+                                    if (!auxList.contains(auxVersionable2)) {
+                                        CompareOutput auxOutput = new CompareOutput(auxAnnotation.fieldTitle(),
+                                                new String[] {
+                                                        ActivityVersionUtil
+                                                                .generateFormattedOutput(auxVersionable1.getOutput()),
+                                                        ActivityVersionUtil
+                                                                .generateFormattedOutput(auxVersionable2.getOutput()) },
+                                                fields[i], new Object[] { auxObject1, auxObject2 }, false, false);
+                                        outputCollection.add(auxOutput);
+                                    }
+                                }
+                            }
+                        }
+                        if (coincidence == 0) {
+                            // Check if the object was added in the
+                            // previous iteration.
+                            if (!auxList.contains(auxVersionable2)) {
+                                CompareOutput auxOutput = new CompareOutput(auxAnnotation.fieldTitle(),
+                                        new String[] { "",
+                                                ActivityVersionUtil
+                                                        .generateFormattedOutput(auxVersionable2.getOutput()) },
+                                        fields[i], new Object[] { null, auxObject2 }, false, false);
+                                outputCollection.add(auxOutput);
+                            }
+                        }
+                    }
+                } else {
+                    output.setStringOutput(new String[] { auxResult1 != null ? auxResult1.toString() : "",
+                            auxResult2 != null ? auxResult2.toString() : "" });
+                    output.setDescriptionOutput(auxAnnotation.fieldTitle());
+                    outputCollection.add(output);
+                }
+            }
+        }
+        return ActivityVersionUtil.groupOutputCollection(outputCollection);
+    }
  
-    
+    public static Map<Long, Map<String, List<CompareOutput>>> compareActivities(List<Long> activityId)
+            throws Exception {
+
+        Map<Long, Map<String, List<CompareOutput>>> listOfActivities = new HashMap<Long, Map<String, List<CompareOutput>>>();
+
+        ArrayList<Long> activityList = new ArrayList(activityId);
+        Iterator<Long> iter = activityList.iterator();
+        while (iter.hasNext()) {
+            Long currentActivity = iter.next();
+            Map<String, List<CompareOutput>> chkPrevous = compareActivities(currentActivity);
+            if (chkPrevous == null) {
```

Comment:
to make the code simple just check if the variable is !null add it to the map, instead of using by null and using continue